instrumentation/kafka-clients/kafka-clients-0.11/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/TracingKafkaUtils.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import static io.opentelemetry.instrumentation.kafka.KafkaSingletons.consumerProcessInstrumenter;
+import static io.opentelemetry.instrumentation.kafka.KafkaSingletons.producerInstrumenter;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.instrumentation.kafka.KafkaHeadersGetter;
+import io.opentelemetry.instrumentation.kafka.KafkaHeadersSetter;
+import java.util.Collection;
+import java.util.Collections;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TracingKafkaUtils {
```

Comment:
Instead of utils-type classes in library instrumentations we usually have `...Tracing` & `...TracingBuilder` classes - check out the [okhttp](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/OkHttpTracing.java) [examples](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/OkHttpTracingBuilder.java).
In case of this PR, you'd have a `KafkaTracing` class that exposes the following methods:
```java
public <K, V> Consumer<K, V> newTracingConsumer(Consumer<K, V> originalConsumer) { 
  // ...
}

public <K, V> Producer<K, V> newTracingProducer(Producer<K, V> originalProducer) { 
  // ...
}
```
(or some very similar ones) and completely encapsulates all actual tracing-related things in package protected classes.
A `KafkaTracingBuilder` could allow registering additional `AttributesExtractor`s for customization (instead of `SpanDecorator`).

Reply:
OK, will try to rewrite to this.
I was just following OpenTracing, and trying to match that.

Reply:
Thanks! We've established our won conventions about writing instrumentations, they might differ (a lot) from the OpenTracing ones.